File: RegexTester/EscapeForm.vb
Imports Microsoft.VisualBasic.CompilerServices
Imports System
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.Runtime.CompilerServices
Imports System.Text.RegularExpressions
Imports System.Windows.Forms

Namespace RegexTester
    <DesignerGenerated> _
    Public Class EscapeForm
        Inherits Form
        ' Methods
        Public Sub New()
            AddHandler MyBase.Load, New EventHandler(AddressOf Me.EscapeForm_Load)
            Me.InitializeComponent
        End Sub

        Private Sub btnOK_Click(ByVal sender As Object, ByVal e As EventArgs)
            If (Me.chkCopyToClipboard.Checked AndAlso (Me.txtResult.TextLength > 0)) Then
                Clipboard.SetText(Me.txtResult.Text)
            End If
            Me.DialogResult = DialogResult.OK
        End Sub

        <DebuggerNonUserCode> _
        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing AndAlso (Not Me.components Is Nothing)) Then
                Me.components.Dispose
            End If
            MyBase.Dispose(disposing)
        End Sub

        Private Sub EscapeForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            Helpers.SetTooltipsAndHelpMessages(Me, Me.ToolTip1, Me.HelpProvider1)
            Me.txtText.Text = Me.Options.RegexText
            Me.lblError.Text = ""
        End Sub

        <DebuggerStepThrough> _
        Private Sub InitializeComponent()
            Me.components = New Container
            Me.Label1 = New Label
            Me.txtText = New TextBox
            Me.radEscape = New RadioButton
            Me.radUnescape = New RadioButton
            Me.Label2 = New Label
            Me.txtResult = New TextBox
            Me.chkCopyToClipboard = New CheckBox
            Me.btnOK = New Button
            Me.btnCancel = New Button
            Me.lblError = New Label
            Me.ToolTip1 = New ToolTip(Me.components)
            Me.HelpProvider1 = New HelpProvider
            Me.SuspendLayout
            Dim point As New Point(12, 9)
            Me.Label1.Location = point
            Me.Label1.Name = "Label1"
            Dim size As New Size(&H25, &H17)
            Me.Label1.Size = size
            Me.Label1.TabIndex = 0
            Me.Label1.Text = "&Text"
            point = New Point(&H37, 9)
            Me.txtText.Location = point
            Me.txtText.Multiline = True
            Me.txtText.Name = "txtText"
            Me.txtText.ScrollBars = ScrollBars.Vertical
            size = New Size(&H228, &H61)
            Me.txtText.Size = size
            Me.txtText.TabIndex = 1
            Me.txtText.Text = "Enter a string that you want to escape or unescape."
            Me.radEscape.AutoSize = True
            Me.radEscape.Checked = True
            point = New Point(&H37, &H70)
            Me.radEscape.Location = point
            Me.radEscape.Name = "radEscape"
            size = New Size(&H3D, &H11)
            Me.radEscape.Size = size
            Me.radEscape.TabIndex = 2
            Me.radEscape.TabStop = True
            Me.radEscape.Text = "&Escape"
            Me.ToolTip1.SetToolTip(Me.radEscape, "Test the Escape command.")
            Me.radEscape.UseVisualStyleBackColor = True
            Me.radUnescape.AutoSize = True
            point = New Point(&H7A, &H70)
            Me.radUnescape.Location = point
            Me.radUnescape.Name = "radUnescape"
            size = New Size(&H4A, &H11)
            Me.radUnescape.Size = size
            Me.radUnescape.TabIndex = 3
            Me.radUnescape.Text = "&Unescape"
            Me.ToolTip1.SetToolTip(Me.radUnescape, "Test the Unescape command.")
            Me.radUnescape.UseVisualStyleBackColor = True
            point = New Point(12, &H89)
            Me.Label2.Location = point
            Me.Label2.Name = "Label2"
            size = New Size(&H25, &H17)
            Me.Label2.Size = size
            Me.Label2.TabIndex = 5
            Me.Label2.Text = "&Result"
            point = New Point(&H37, &H89)
            Me.txtResult.Location = point
            Me.txtResult.Multiline = True
            Me.txtResult.Name = "txtResult"
            Me.txtResult.ReadOnly = True
            Me.txtResult.ScrollBars = ScrollBars.Vertical
            size = New Size(&H228, &H5E)
            Me.txtResult.Size = size
            Me.txtResult.TabIndex = 6
            Me.txtResult.Text = "The result of the Escape or Unescape method."
            Me.chkCopyToClipboard.AutoSize = True
            Me.chkCopyToClipboard.Checked = True
            Me.chkCopyToClipboard.CheckState = CheckState.Checked
            point = New Point(&H1D7, &H72)
            Me.chkCopyToClipboard.Location = point
            Me.chkCopyToClipboard.Name = "chkCopyToClipboard"
            size = New Size(&H89, &H11)
            Me.chkCopyToClipboard.Size = size
            Me.chkCopyToClipboard.TabIndex = 4
            Me.chkCopyToClipboard.Text = "C&opy result to Clipboard"
            Me.ToolTip1.SetToolTip(Me.chkCopyToClipboard, "If selected, the result is copied to the Clipboard hen the user clicks on the OK button.")
            Me.chkCopyToClipboard.UseVisualStyleBackColor = True
            Me.btnOK.DialogResult = DialogResult.OK
            point = New Point(&H1C3, &HED)
            Me.btnOK.Location = point
            Me.btnOK.Name = "btnOK"
            size = New Size(&H4B, &H17)
            Me.btnOK.Size = size
            Me.btnOK.TabIndex = 8
            Me.btnOK.Text = "OK"
            Me.ToolTip1.SetToolTip(Me.btnOK, "Close the dialog box and optionally copy the result to the Clipboard.")
            Me.btnOK.UseVisualStyleBackColor = True
            Me.btnCancel.DialogResult = DialogResult.Cancel
            point = New Point(&H214, &HED)
            Me.btnCancel.Location = point
            Me.btnCancel.Name = "btnCancel"
            size = New Size(&H4B, &H17)
            Me.btnCancel.Size = size
            Me.btnCancel.TabIndex = 9
            Me.btnCancel.Text = "Cancel"
            Me.ToolTip1.SetToolTip(Me.btnCancel, "Close the dialog box without copying the result to the Clipboard.")
            Me.btnCancel.UseVisualStyleBackColor = True
            Me.lblError.AutoSize = True
            Me.lblError.ForeColor = Color.Red
            point = New Point(&H34, &HED)
            Me.lblError.Location = point
            Me.lblError.Name = "lblError"
            size = New Size(&H27, 13)
            Me.lblError.Size = size
            Me.lblError.TabIndex = 7
            Me.lblError.Text = "lblError"
            Me.AcceptButton = Me.btnOK
            Dim ef As New SizeF(6!, 13!)
            Me.AutoScaleDimensions = ef
            Me.AutoScaleMode = AutoScaleMode.Font
            Me.CancelButton = Me.btnCancel
            size = New Size(&H26B, 270)
            Me.ClientSize = size
            Me.Controls.Add(Me.lblError)
            Me.Controls.Add(Me.btnCancel)
            Me.Controls.Add(Me.btnOK)
            Me.Controls.Add(Me.chkCopyToClipboard)
            Me.Controls.Add(Me.txtResult)
            Me.Controls.Add(Me.Label2)
            Me.Controls.Add(Me.radUnescape)
            Me.Controls.Add(Me.radEscape)
            Me.Controls.Add(Me.txtText)
            Me.Controls.Add(Me.Label1)
            Me.Name = "EscapeForm"
            Me.Text = "Escape Command"
            Me.ResumeLayout(False)
            Me.PerformLayout
        End Sub

        Private Sub radUnescape_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            Me.txtText.Text = ""
        End Sub

        Private Sub txtText_TextChanged(ByVal sender As Object, ByVal e As EventArgs)
            Try 
                If Me.radEscape.Checked Then
                    Me.txtResult.Text = Regex.Escape(Me.txtText.Text)
                Else
                    Me.txtResult.Text = Regex.Unescape(Me.txtText.Text)
                End If
                Me.lblError.Text = ""
            Catch exception1 As Exception
                ProjectData.SetProjectError(exception1)
                Dim exception As Exception = exception1
                Me.lblError.Text = exception.Message
                ProjectData.ClearProjectError
            End Try
        End Sub


        ' Properties
        Friend Overridable Property Label1 As Label
            Get
                Return Me._Label1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As Label)
                Me._Label1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property txtText As TextBox
            Get
                Return Me._txtText
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As TextBox)
                If (Not Me._txtText Is Nothing) Then
                    RemoveHandler Me._txtText.TextChanged, New EventHandler(AddressOf Me.txtText_TextChanged)
                End If
                Me._txtText = WithEventsValue
                If (Not Me._txtText Is Nothing) Then
                    AddHandler Me._txtText.TextChanged, New EventHandler(AddressOf Me.txtText_TextChanged)
                End If
            End Set
        End Property

        Friend Overridable Property radEscape As RadioButton
            Get
                Return Me._radEscape
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As RadioButton)
                If (Not Me._radEscape Is Nothing) Then
                    RemoveHandler Me._radEscape.CheckedChanged, New EventHandler(AddressOf Me.radUnescape_CheckedChanged)
                End If
                Me._radEscape = WithEventsValue
                If (Not Me._radEscape Is Nothing) Then
                    AddHandler Me._radEscape.CheckedChanged, New EventHandler(AddressOf Me.radUnescape_CheckedChanged)
                End If
            End Set
        End Property

        Friend Overridable Property radUnescape As RadioButton
            Get
                Return Me._radUnescape
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As RadioButton)
                If (Not Me._radUnescape Is Nothing) Then
                    RemoveHandler Me._radUnescape.CheckedChanged, New EventHandler(AddressOf Me.radUnescape_CheckedChanged)
                End If
                Me._radUnescape = WithEventsValue
                If (Not Me._radUnescape Is Nothing) Then
                    AddHandler Me._radUnescape.CheckedChanged, New EventHandler(AddressOf Me.radUnescape_CheckedChanged)
                End If
            End Set
        End Property

        Friend Overridable Property Label2 As Label
            Get
                Return Me._Label2
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As Label)
                Me._Label2 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property txtResult As TextBox
            Get
                Return Me._txtResult
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As TextBox)
                Me._txtResult = WithEventsValue
            End Set
        End Property

        Friend Overridable Property chkCopyToClipboard As CheckBox
            Get
                Return Me._chkCopyToClipboard
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As CheckBox)
                Me._chkCopyToClipboard = WithEventsValue
            End Set
        End Property

        Friend Overridable Property btnOK As Button
            Get
                Return Me._btnOK
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As Button)
                If (Not Me._btnOK Is Nothing) Then
                    RemoveHandler Me._btnOK.Click, New EventHandler(AddressOf Me.btnOK_Click)
                End If
                Me._btnOK = WithEventsValue
                If (Not Me._btnOK Is Nothing) Then
                    AddHandler Me._btnOK.Click, New EventHandler(AddressOf Me.btnOK_Click)
                End If
            End Set
        End Property

        Friend Overridable Property btnCancel As Button
            Get
                Return Me._btnCancel
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As Button)
                Me._btnCancel = WithEventsValue
            End Set
        End Property

        Friend Overridable Property lblError As Label
            Get
                Return Me._lblError
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As Label)
                Me._lblError = WithEventsValue
            End Set
        End Property

        Friend Overridable Property ToolTip1 As ToolTip
            Get
                Return Me._ToolTip1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As ToolTip)
                Me._ToolTip1 = WithEventsValue
            End Set
        End Property

        Friend Overridable Property HelpProvider1 As HelpProvider
            Get
                Return Me._HelpProvider1
            End Get
            <MethodImpl(MethodImplOptions.Synchronized)> _
            Set(ByVal WithEventsValue As HelpProvider)
                Me._HelpProvider1 = WithEventsValue
            End Set
        End Property


        ' Fields
        Private components As IContainer
        <AccessedThroughProperty("Label1")> _
        Private _Label1 As Label
        <AccessedThroughProperty("txtText")> _
        Private _txtText As TextBox
        <AccessedThroughProperty("radEscape")> _
        Private _radEscape As RadioButton
        <AccessedThroughProperty("radUnescape")> _
        Private _radUnescape As RadioButton
        <AccessedThroughProperty("Label2")> _
        Private _Label2 As Label
        <AccessedThroughProperty("txtResult")> _
        Private _txtResult As TextBox
        <AccessedThroughProperty("chkCopyToClipboard")> _
        Private _chkCopyToClipboard As CheckBox
        <AccessedThroughProperty("btnOK")> _
        Private _btnOK As Button
        <AccessedThroughProperty("btnCancel")> _
        Private _btnCancel As Button
        <AccessedThroughProperty("lblError")> _
        Private _lblError As Label
        <AccessedThroughProperty("ToolTip1")> _
        Private _ToolTip1 As ToolTip
        <AccessedThroughProperty("HelpProvider1")> _
        Private _HelpProvider1 As HelpProvider
        Public Options As ProjectOptions
    End Class
End Namespace

File: RegexTester/EscapeForm.cs
namespace RegexTester
{
    using Microsoft.VisualBasic.CompilerServices;
    using System;
    using System.ComponentModel;
    using System.Diagnostics;
    using System.Drawing;
    using System.Runtime.CompilerServices;
    using System.Text.RegularExpressions;
    using System.Windows.Forms;

    [DesignerGenerated]
    public class EscapeForm : Form
    {
        private IContainer components;
        [AccessedThroughProperty("Label1")]
        private Label _Label1;
        [AccessedThroughProperty("txtText")]
        private TextBox _txtText;
        [AccessedThroughProperty("radEscape")]
        private RadioButton _radEscape;
        [AccessedThroughProperty("radUnescape")]
        private RadioButton _radUnescape;
        [AccessedThroughProperty("Label2")]
        private Label _Label2;
        [AccessedThroughProperty("txtResult")]
        private TextBox _txtResult;
        [AccessedThroughProperty("chkCopyToClipboard")]
        private CheckBox _chkCopyToClipboard;
        [AccessedThroughProperty("btnOK")]
        private Button _btnOK;
        [AccessedThroughProperty("btnCancel")]
        private Button _btnCancel;
        [AccessedThroughProperty("lblError")]
        private Label _lblError;
        [AccessedThroughProperty("ToolTip1")]
        private ToolTip _ToolTip1;
        [AccessedThroughProperty("HelpProvider1")]
        private HelpProvider _HelpProvider1;
        public ProjectOptions Options;

        public EscapeForm()
        {
            base.Load += new EventHandler(this.EscapeForm_Load);
            this.InitializeComponent();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (this.chkCopyToClipboard.Checked && (this.txtResult.TextLength > 0))
            {
                Clipboard.SetText(this.txtResult.Text);
            }
            this.DialogResult = DialogResult.OK;
        }

        [DebuggerNonUserCode]
        protected override void Dispose(bool disposing)
        {
            if (disposing && (this.components != null))
            {
                this.components.Dispose();
            }
            base.Dispose(disposing);
        }

        private void EscapeForm_Load(object sender, EventArgs e)
        {
            Helpers.SetTooltipsAndHelpMessages(this, this.ToolTip1, this.HelpProvider1);
            this.txtText.Text = this.Options.RegexText;
            this.lblError.Text = "";
        }

        [DebuggerStepThrough]
        private void InitializeComponent()
        {
            this.components = new System.ComponentModel.Container();
            this.Label1 = new Label();
            this.txtText = new TextBox();
            this.radEscape = new RadioButton();
            this.radUnescape = new RadioButton();
            this.Label2 = new Label();
            this.txtResult = new TextBox();
            this.chkCopyToClipboard = new CheckBox();
            this.btnOK = new Button();
            this.btnCancel = new Button();
            this.lblError = new Label();
            this.ToolTip1 = new ToolTip(this.components);
            this.HelpProvider1 = new HelpProvider();
            this.SuspendLayout();
            Point point = new Point(12, 9);
            this.Label1.Location = point;
            this.Label1.Name = "Label1";
            Size size = new Size(0x25, 0x17);
            this.Label1.Size = size;
            this.Label1.TabIndex = 0;
            this.Label1.Text = "&Text";
            point = new Point(0x37, 9);
            this.txtText.Location = point;
            this.txtText.Multiline = true;
            this.txtText.Name = "txtText";
            this.txtText.ScrollBars = ScrollBars.Vertical;
            size = new Size(0x228, 0x61);
            this.txtText.Size = size;
            this.txtText.TabIndex = 1;
            this.txtText.Text = "Enter a string that you want to escape or unescape.";
            this.radEscape.AutoSize = true;
            this.radEscape.Checked = true;
            point = new Point(0x37, 0x70);
            this.radEscape.Location = point;
            this.radEscape.Name = "radEscape";
            size = new Size(0x3d, 0x11);
            this.radEscape.Size = size;
            this.radEscape.TabIndex = 2;
            this.radEscape.TabStop = true;
            this.radEscape.Text = "&Escape";
            this.ToolTip1.SetToolTip(this.radEscape, "Test the Escape command.");
            this.radEscape.UseVisualStyleBackColor = true;
            this.radUnescape.AutoSize = true;
            point = new Point(0x7a, 0x70);
            this.radUnescape.Location = point;
            this.radUnescape.Name = "radUnescape";
            size = new Size(0x4a, 0x11);
            this.radUnescape.Size = size;
            this.radUnescape.TabIndex = 3;
            this.radUnescape.Text = "&Unescape";
            this.ToolTip1.SetToolTip(this.radUnescape, "Test the Unescape command.");
            this.radUnescape.UseVisualStyleBackColor = true;
            point = new Point(12, 0x89);
            this.Label2.Location = point;
            this.Label2.Name = "Label2";
            size = new Size(0x25, 0x17);
            this.Label2.Size = size;
            this.Label2.TabIndex = 5;
            this.Label2.Text = "&Result";
            point = new Point(0x37, 0x89);
            this.txtResult.Location = point;
            this.txtResult.Multiline = true;
            this.txtResult.Name = "txtResult";
            this.txtResult.ReadOnly = true;
            this.txtResult.ScrollBars = ScrollBars.Vertical;
            size = new Size(0x228, 0x5e);
            this.txtResult.Size = size;
            this.txtResult.TabIndex = 6;
            this.txtResult.Text = "The result of the Escape or Unescape method.";
            this.chkCopyToClipboard.AutoSize = true;
            this.chkCopyToClipboard.Checked = true;
            this.chkCopyToClipboard.CheckState = CheckState.Checked;
            point = new Point(0x1d7, 0x72);
            this.chkCopyToClipboard.Location = point;
            this.chkCopyToClipboard.Name = "chkCopyToClipboard";
            size = new Size(0x89, 0x11);
            this.chkCopyToClipboard.Size = size;
            this.chkCopyToClipboard.TabIndex = 4;
            this.chkCopyToClipboard.Text = "C&opy result to Clipboard";
            this.ToolTip1.SetToolTip(this.chkCopyToClipboard, "If selected, the result is copied to the Clipboard hen the user clicks on the OK button.");
            this.chkCopyToClipboard.UseVisualStyleBackColor = true;
            this.btnOK.DialogResult = DialogResult.OK;
            point = new Point(0x1c3, 0xed);
            this.btnOK.Location = point;
            this.btnOK.Name = "btnOK";
            size = new Size(0x4b, 0x17);
            this.btnOK.Size = size;
            this.btnOK.TabIndex = 8;
            this.btnOK.Text = "OK";
            this.ToolTip1.SetToolTip(this.btnOK, "Close the dialog box and optionally copy the result to the Clipboard.");
            this.btnOK.UseVisualStyleBackColor = true;
            this.btnCancel.DialogResult = DialogResult.Cancel;
            point = new Point(0x214, 0xed);
            this.btnCancel.Location = point;
            this.btnCancel.Name = "btnCancel";
            size = new Size(0x4b, 0x17);
            this.btnCancel.Size = size;
            this.btnCancel.TabIndex = 9;
            this.btnCancel.Text = "Cancel";
            this.ToolTip1.SetToolTip(this.btnCancel, "Close the dialog box without copying the result to the Clipboard.");
            this.btnCancel.UseVisualStyleBackColor = true;
            this.lblError.AutoSize = true;
            this.lblError.ForeColor = Color.Red;
            point = new Point(0x34, 0xed);
            this.lblError.Location = point;
            this.lblError.Name = "lblError";
            size = new Size(0x27, 13);
            this.lblError.Size = size;
            this.lblError.TabIndex = 7;
            this.lblError.Text = "lblError";
            this.AcceptButton = this.btnOK;
            SizeF ef = new SizeF(6f, 13f);
            this.AutoScaleDimensions = ef;
            this.AutoScaleMode = AutoScaleMode.Font;
            this.CancelButton = this.btnCancel;
            size = new Size(0x26b, 270);
            this.ClientSize = size;
            this.Controls.Add(this.lblError);
            this.Controls.Add(this.btnCancel);
            this.Controls.Add(this.btnOK);
            this.Controls.Add(this.chkCopyToClipboard);
            this.Controls.Add(this.txtResult);
            this.Controls.Add(this.Label2);
            this.Controls.Add(this.radUnescape);
            this.Controls.Add(this.radEscape);
            this.Controls.Add(this.txtText);
            this.Controls.Add(this.Label1);
            this.Name = "EscapeForm";
            this.Text = "Escape Command";
            this.ResumeLayout(false);
            this.PerformLayout();
        }

        private void radUnescape_CheckedChanged(object sender, EventArgs e)
        {
            this.txtText.Text = "";
        }

        private void txtText_TextChanged(object sender, EventArgs e)
        {
            try
            {
                if (this.radEscape.Checked)
                {
                    this.txtResult.Text = Regex.Escape(this.txtText.Text);
                }
                else
                {
                    this.txtResult.Text = Regex.Unescape(this.txtText.Text);
                }
                this.lblError.Text = "";
            }
            catch (Exception exception1)
            {
                ProjectData.SetProjectError(exception1);
                Exception exception = exception1;
                this.lblError.Text = exception.Message;
                ProjectData.ClearProjectError();
            }
        }

        internal virtual Label Label1
        {
            get
            {
                return this._Label1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label1 = value;
            }
        }

        internal virtual TextBox txtText
        {
            get
            {
                return this._txtText;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (this._txtText != null)
                {
                    this._txtText.TextChanged -= new EventHandler(this.txtText_TextChanged);
                }
                this._txtText = value;
                if (this._txtText != null)
                {
                    this._txtText.TextChanged += new EventHandler(this.txtText_TextChanged);
                }
            }
        }

        internal virtual RadioButton radEscape
        {
            get
            {
                return this._radEscape;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (this._radEscape != null)
                {
                    this._radEscape.CheckedChanged -= new EventHandler(this.radUnescape_CheckedChanged);
                }
                this._radEscape = value;
                if (this._radEscape != null)
                {
                    this._radEscape.CheckedChanged += new EventHandler(this.radUnescape_CheckedChanged);
                }
            }
        }

        internal virtual RadioButton radUnescape
        {
            get
            {
                return this._radUnescape;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (this._radUnescape != null)
                {
                    this._radUnescape.CheckedChanged -= new EventHandler(this.radUnescape_CheckedChanged);
                }
                this._radUnescape = value;
                if (this._radUnescape != null)
                {
                    this._radUnescape.CheckedChanged += new EventHandler(this.radUnescape_CheckedChanged);
                }
            }
        }

        internal virtual Label Label2
        {
            get
            {
                return this._Label2;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._Label2 = value;
            }
        }

        internal virtual TextBox txtResult
        {
            get
            {
                return this._txtResult;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._txtResult = value;
            }
        }

        internal virtual CheckBox chkCopyToClipboard
        {
            get
            {
                return this._chkCopyToClipboard;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._chkCopyToClipboard = value;
            }
        }

        internal virtual Button btnOK
        {
            get
            {
                return this._btnOK;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                if (this._btnOK != null)
                {
                    this._btnOK.Click -= new EventHandler(this.btnOK_Click);
                }
                this._btnOK = value;
                if (this._btnOK != null)
                {
                    this._btnOK.Click += new EventHandler(this.btnOK_Click);
                }
            }
        }

        internal virtual Button btnCancel
        {
            get
            {
                return this._btnCancel;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._btnCancel = value;
            }
        }

        internal virtual Label lblError
        {
            get
            {
                return this._lblError;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._lblError = value;
            }
        }

        internal virtual ToolTip ToolTip1
        {
            get
            {
                return this._ToolTip1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._ToolTip1 = value;
            }
        }

        internal virtual HelpProvider HelpProvider1
        {
            get
            {
                return this._HelpProvider1;
            }
            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                this._HelpProvider1 = value;
            }
        }
    }
}

